extensions/wikia/ImageReview/ImageListGetter.class.php

```diff
@@ -0,0 +1,368 @@
+
+		$reviewStart = wfTimestamp( TS_DB, $timestamp );
+		$imageIter = $this->getLatestImageIter( $order, $state );
+		while ( $softNum && $hardNum && $row = $imageIter() ) {
+			$hardNum--;
+
+			// Grab this image now to eliminate any race conditions with other reviewers
+			if ( !$this->assignImageToUser( $reviewStart, $state, $row ) ) {
+				// If we failed to assign this image to ourselves, try the next
+				continue;
+			}
+
+			$imageInfo = $this->getImageData( $row );
+
+			// If we failed to load this image, remove it from the review queue
+			if ( !empty( $imageInfo['error'] ) ) {
+				$deleteFromQueueList[] = [
+					'wiki_id' => $row->wiki_id,
+					'page_id' => $row->page_id,
+					'reason' => $imageInfo['error'],
+				];
+				continue;
+			}
+
+			// If this is an icon, put it in a separate queue
+			if ( strpos( 'ico', $imageInfo['extension'] ) ) {
+				$iconsWhere[] = "(wiki_id = {$row->wiki_id} and page_id = {$row->page_id})";
+				continue;
+			}
+
+			$imageList[] = $imageInfo;
+			$softNum--;
+		}
+
+		// Remove invalid images
+		$this->createDeleteFromQueueTask( $deleteFromQueueList );
+
+		// Move icons to different state
+		$this->moveImagesToIcoStateFromWhere( $iconsWhere );
+
+		// Return valid images list
+		WikiaLogger::instance()->info( 'ImageReviewLog', [
+			'method' => __METHOD__,
+			'message' => 'Fetched ' . count( $imageList ) . ' new images',
+		] );
+
+		return $imageList;
+	}
+
+	/**
+	 * @param $timestamp
+	 * @return bool
+	 * @throws Exception
+	 * @throws MWException
+	 */
+	private function previouslyViewedImageList( $timestamp ) {
+		$db = $this->getDatawareDB();
+		return ( new WikiaSQL() )
+			->SELECT( '1' )
+			->FROM( 'image_review' )
+			->WHERE( 'reviewer_id' )->EQUAL_TO( $this->userId )
+			->AND_( 'review_start' )->EQUAL_TO( wfTimestamp( TS_DB, $timestamp ) )
+			->LIMIT( 1 )
+			->run( $db, function( ResultWrapper $result ) {
+				return !empty( $result->fetchRow() );
+			} );
+	}
+
+	/**
+	 * Get image list from reviewer id based on the timestamp
+	 * Note: Does NOT update image state
+	 *
+	 * @param integer $timestamp review_end
+	 * @return array images
+	 */
+	public function refetchImageListByTimestamp( $timestamp ) {
+		wfProfileIn( __METHOD__ );
+
+		$db = $this->getDatawareDB( DB_SLAVE );
+
+		$review_start = wfTimestamp( TS_DB, $timestamp );
+
+		$result = $db->select(
+			[  'image_review'  ],
+			[  'wiki_id, page_id, state, flags, priority'  ],
+			[
+				'review_start'	=> $review_start,
+				'reviewer_id'	=> $this->userId
+			],
+			__METHOD__,
+			[
+				'ORDER BY' => 'priority desc, last_edited desc',
+				'LIMIT' => self::LIMIT_IMAGES
+			]
+		);
+
+		$imageList = [];
+		while ( $row = $db->fetchObject( $result ) ) {
+			$img = ImagesService::getImageSrc( $row->wiki_id, $row->page_id );
+			$wikiRow = WikiFactory::getWikiByID( $row->wiki_id );
+			$tmp = [
+				'wikiId' 	=> $row->wiki_id,
+				'pageId' 	=> $row->page_id,
+				'state' 	=> $row->state,
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "state" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)